daemon/api.go

```diff
@@ -367,6 +374,59 @@ func webify(result map[string]interface{}, resource string) map[string]interface
 	return result
 }
 
+func searchStore(c *Command, r *http.Request) Response {
+	route := c.d.router.Get(snapCmd.Path)
+	if route == nil {
+		return InternalError("router can't find route for snaps")
+	}
+
+	query := r.URL.Query()
+
+	auther, err := c.d.auther(r)
+	if err != nil && err != auth.ErrInvalidAuth {
+		return InternalError("%v", err)
+	}
+
+	remoteRepo := newRemoteRepo()
+	found, err := remoteRepo.FindSnaps(query.Get("q"), query.Get("channel"), auther)
+	if err != nil {
+		status := http.StatusInternalServerError
+		if e, ok := err.(*store.HTTPError); ok && e != nil {
+			status = e.StatusCode
+		}
+
+		return &resp{
+			Type: ResponseTypeError,
+			Result: &errorResult{
+				Message: err.Error(),
```

Comment:
As a side effect of the logic above, I'm wondering a bit if the error messages being returned from FindSnaps are sane or if we can get things such as "Not Found" here for such a 404 case, which would be very confusing because it would sound like the snap was not found when in fact we could not talk to the server.